### PR TITLE
lamexp: Add version 4.17.2188

### LINKED ISSUE #1907: [Request] LameXP - Audio Encoder Front-End
**Website**
[http://lamexp.sourceforge.net/](http://lamexp.sourceforge.net/)
**Downloads**
[https://sourceforge.net/projects/lamexp/files/latest/download](https://sourceforge.net/projects/lamexp/files/latest/download
)
[https://sourceforge.net/projects/lamexp/files/](https://sourceforge.net/projects/lamexp/files/)

### PR DESCRIPTION
**Pull request by niheaven**
Please merge after https://github.com/lukesampson/scoop/pull/3263, close #1907

### CODE REVIEW

File: bucket/lamexp.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    "depends": "nero-aac",
+    "pre_install": [
+        "Rename-Item \"$dir\\LameXP.exe\" \"$dir\\LameXP-Portable.exe\"",
+        "$ini = \"LameXP-Portable.ini\"",
+        "If (!(Test-Path \"$persist_dir\\$ini\")) {",
+        "   New-Item \"$dir\\$ini\" | Out-Null",
+        "} Else {",
+        "   $version -match \"(\\d+).(\\d+).(\\d+)\" | Out-Null",
+        "   $cleanVersion = $Matches[1] + $Matches[2] + '0' + $Matches[3]",
+        "   (Get-Content \"$persist_dir\\$ini\") | Foreach-Object { $_ -replace \"LameXP_\\d+\", \"LameXP_$cleanVersion\" } | Set-Content \"$persist_dir\\$ini\"",
```
Comment (Ash258) [inline bucket/lamexp.json]: For each is not needed. Replace is called on all array elements, when usead on array
```suggestion
        "   (Get-Content \"$persist_dir\\$ini\") -replace 'LameXP_\\d+', \"LameXP_$cleanVersion\" | Set-Content \"$persist_dir\\$ini\"",
```

File: bucket/lamexp.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        "},",
+        "$aac = versiondir 'nero-aac' 'current'",
+        "If (Test-Path $aac) {",
+        "   info \"Adding Nero AAC Codec Support\"",
+        "   Copy-Item \"$aac\\win32\\*.exe\" $dir",
```
Comment (Ash258) [inline bucket/lamexp.json]: Not quoeted dir variable
```suggestion
        "   Copy-Item \"$aac\\win32\\*.exe\" \"$dir\"",
```

File: bucket/lamexp.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        "$ini = \"LameXP-Portable.ini\"",
+        "If (!(Test-Path \"$persist_dir\\$ini\")) {",
+        "   New-Item \"$dir\\$ini\" | Out-Null",
+        "} Else {",
+        "   $version -match \"(\\d+).(\\d+).(\\d+)\" | Out-Null",
+        "   $cleanVersion = $Matches[1] + $Matches[2] + '0' + $Matches[3]",
+        "   (Get-Content \"$persist_dir\\$ini\") | Foreach-Object { $_ -replace \"LameXP_\\d+\", \"LameXP_$cleanVersion\" } | Set-Content \"$persist_dir\\$ini\"",
+        "},",
+        "$aac = versiondir 'nero-aac' 'current'",
+        "If (Test-Path $aac) {",
```
Comment (Ash258) [inline bucket/lamexp.json]: All path variables, which could contains spaces needs to be quoted.
```suggestion
        "If (Test-Path \"$aac\") {",
```

**Comment by niheaven** [comment]
Thanks, I've forgotten the case that `$scoopdir` may contain spaces.

**Comment by pedrogarcia** [comment]
@niheaven Can you add a dependency of qaac for this package as well? Like the nero-aac package, the qaac package (already exists in the scoop main repo) just needs to be copied to the lamexp folder.

**Comment by niheaven** [comment]
@pedrogarcia Okay, I'll add an alternative lamexp-qaac in versions bucket, and maybe nero-aac and qaac is exclusive?

**Comment by pedrogarcia** [comment]
@niheaven Nope, not exclusive. Lamexp automatically uses qaac if it is able. Just copy the contents of package qaac in the lamexp folder, together with nero-aac.

**Comment by niheaven** [comment]
Please see `versions/lamexp-qaac`(https://github.com/scoopinstaller/versions/pull/128).

You could choose the one you like, I don't think get two AAC encoder into one place is a good idea, so maybe two versions is better.

**Comment by pedrogarcia** [comment]
Yep, maybe having two versions is better, because qaac also depends on
iTunes I think.

**Comment by niheaven** [comment]
`QAAC` needs so called `Apple Application Support` in QuickTime/iTunes, and which is not included in iTunes Windows Store version... That wastes me some time to check why qaac doesn't work after copying...

**Pull request merged**